File: Tutorial3-1Scope1/src/main/java/com/example/tutorial3scope1/MyApplication.java
package com.example.tutorial3scope1;

import android.app.Application;

import com.example.tutorial3scope1.di.DaggerUserComponent;
import com.example.tutorial3scope1.di.UserComponent;


public class MyApplication extends Application {

    private UserComponent userComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        userComponent = DaggerUserComponent.create();
    }


    public UserComponent getUserComponent() {
        return userComponent;
    }
}

File: Tutorial3-1Scope1/src/main/java/com/example/tutorial3scope1/MyApplication.kt
package com.example.tutorial3scope1

import android.app.Application
import com.example.tutorial3scope1.di.DaggerUserComponent
import com.example.tutorial3scope1.di.UserComponent

class MyApplication : Application() {

    lateinit var userComponent: UserComponent
        private set

    override fun onCreate() {
        super.onCreate()
        userComponent = DaggerUserComponent.create()
    }

}